### ISSUE TITLE
"Undirected" mode for shortest paths with directed graphs

### ISSUE THREAD

**Issue by etiennedeg** [JuliaGraphs/Graphs.jl#12]
This is a repost of [this issue](https://github.com/sbromberger/LightGraphs.jl/issues/1584) on the old repo.

---

Hi,
This is a follow up of [this discussion](https://discourse.julialang.org/t/help-with-lightgraphs-jl/66663).
Is there a way to efficiently find shortest paths in a directed graph while ignoring the direction of the edges ? (so without casting to a directed graph)
R seems to have an "undirected" mode for this.

---

Having thought more on this, better than a keyword in functions, it would be practical to have a new graph type that would be a wrapper around a directed graph, with redefinition of accessers  to make it an undirected graph (a sort of graph view). This would avoid the heavy conversion to an undirected graph.

**Comment by jpfairbanks** [comment]
Yes I agree, solving this at the type level with a wrapper type that implements the abstract undirected graph API is the right move here. We don't want to add a kwarg for every function to add "undirected mode" just wrap it once and be done
